src/core/speedy-media-source.js

```diff
@@ -710,3 +712,102 @@ class SpeedyBitmapMediaSource extends SpeedyMediaSource
         return new SpeedyBitmapMediaSource(PRIVATE_TOKEN)._load(bitmap);
     }
 }
+
+/**
+ * ImageData media source:
+ * a wrapper around ImageData
+ */
+class SpeedyImageDataMediaSource extends SpeedyMediaSource {
+    /**
+     * @private Constructor
+     * @param {symbol} token
+     */
+    constructor(token) {
+        super(token);
+        this._data = null;
```

Comment:
We can annotate the type here for consistency, as in `/** @type {ImageData} image data */`